doc/parrot.html

```diff
@@ -306,6 +306,56 @@ <h2 id="nameresolution">Name Resolution<a class="sectionlink" href="#nameresolut
 (see contents of /chirp/server.nd.edu/mix/1523.data)
 </code>
 
+<h2 id="namespaces">Mount Namespaces<a class="sectionlink" href="#namespaces" title="Link to this section.">&#x21d7;</a></h2>
+
+<p>Mount entries in Parrot are organized into hierarchical, reference counted namespaces.
+A fresh instance of Parrot puts all process into a single, global namespace.
+This means that normally, all processes running under a Parrot instance see the same
+view of the filesystem. More concretely, all processes share the same mount list specified
+by the <tt>-m</tt> and <tt>-M</tt> options. The <tt>parrot_mount</tt> command allows programs to edit this
+mount list.</p>
+<p>Processes are also free to fork their mount namespaces with the Parrot-specific
+<tt>parrot_fork_namespace</tt> syscall. Whenever a process forks its namespace, its child
+will inherit a reference to its parent’s mount namespace. Mount namespace changes are
+visible to processes subject to the following rules. Given a process <em>P</em> in namespace <em>N</em>,</p>
+<ul>
+<li><em>P</em> may add/remove mount entries in <em>N</em>.</li>
+<li>any other process with a reference to <em>N</em>, will see changes by <em>P</em>.</li>
+<li>suppose another process <em>P’</em> forks <em>N</em>, so that <em>P’</em> is in namespace <em>N’</em>, then
+<ul>
+<li>any changes to <em>N’</em> <strong>are not</strong> visible in <em>N</em>.</li>
+<li>any changes to <em>N</em> <strong>are</strong> visible in <em>N’</em>.</li>
+</ul>
+</li>
+<li>when adding a mount entry to <em>N</em>, the redirect is resolved in the parent namespace of <em>N</em>.</li>
+</ul>
+<p>When Parrot starts, the global mount namespace has no parent, so mounts passed in are
+resolved in the host filesystem. Unless the <tt>--dynamic-mounts</tt> flag is passed,
+Parrot <strong>seals</strong> the global mount namespace before running the tracee by making a new
```

Comment:
Tim,
These mount names have lexical scope, as every time a new parrot_namespace is called, a new scope is created, with the name bindings lasting until the process ends. So they seem to me proper _closures_, and I would call them as such.